[BUG] Build failure with Clang 5 -Wpedantic -Werror and #warning

**Describe the bug**
This library won't compile with Clang 5, but do with all other version of clang.

Here's the error the compiler gives:

```
/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/include -I/home/runner/work/kangaru/kangaru/vcpkg/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/src -I/home/runner/work/kangaru/kangaru/vcpkg/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/src/../include -fPIC  -std=c++11  -Wall  -Wextra  -Wshadow  -pedantic  -pedantic-errors  -Wshorten-64-to-32  -fstrict-aliasing  -Wno-deprecated-declarations  -Wstrict-aliasing  -Wthread-safety -g -MD -MT src/CMakeFiles/benchmark.dir/sysinfo.cc.o -MF src/CMakeFiles/benchmark.dir/sysinfo.cc.o.d -o src/CMakeFiles/benchmark.dir/sysinfo.cc.o -c /home/runner/work/kangaru/kangaru/vcpkg/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/src/sysinfo.cc
  /home/runner/work/kangaru/kangaru/vcpkg/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/src/sysinfo.cc:447:2: error: #warning is a language extension [-Werror,-Wpedantic]
  #warning "HOST_NAME_MAX not defined. using 64"
   ^
  /home/runner/work/kangaru/kangaru/vcpkg/buildtrees/benchmark/src/278e00fdbd-51ea7c4262.clean/src/sysinfo.cc:447:2: warning: "HOST_NAME_MAX not defined. using 64" [-W#warnings]
  1 warning and 1 error generated.
```

It seems benchmark uses a preprocessor extension to give a warning, but having `-Wpedantic -Werror` makes the warning of using the extension an error.

This doesn't happen with all other version of clang.

**System**
Which OS, compiler, and compiler version are you using:
  - OS:  Linux
  - Compiler and version: Clang 5

**To reproduce**
Steps to reproduce the behavior:
1. checkout `v1.5.5`
2. Run `cmake .. -DBENCHMARK_ENABLE_TESTING=OFF -DCMAKE_CXX_COMPILER=clang++-5`
3. Run ninja or make

**Expected behavior**
The library should build correctly

**Additional context**
It was found on github actions to test one of my libraries using Clang 5. [Here's the full log](https://github.com/gracicot/kangaru/runs/3325020925?check_suite_focus=true#step:7:150).

It was installed using vcpkg. The original bug report is here: microsoft/vcpkg#19588

--- a/src/sysinfo.cc
+++ b/src/sysinfo.cc
@@ -444,7 +444,7 @@ std::string GetSystemName() {
 #elif defined(BENCHMARK_OS_RTEMS)
 #define HOST_NAME_MAX 256
 #else
-#warning "HOST_NAME_MAX not defined. using 64"
+#pragma message("HOST_NAME_MAX not defined. using 64")
 #define HOST_NAME_MAX 64
 #endif
 #endif // def HOST_NAME_MAX

